src/pydap/responses/dods.py

```diff
@@ -108,12 +126,21 @@ def _sequencetype(var):
                         padded.append(length + (-length % 4))
                     out.append(value)
                 record = out
-                dtype = ','.join(types).format(*padded)
-
-            if dtype not in cache:
-                cache[dtype] = np.zeros((1,), dtype=dtype)
-            cache[dtype][:] = tuple(record)
-            yield cache[dtype].tostring()
+                DAP2_dtype_str = ','.join(DAP2_types).format(*padded)
+
+            if DAP2_dtype_str not in cache:
+                # Remember that DAP2_dtype is a (possibly composite)
```

Comment:
This was sufficiently confusing and I thought anyone else would be as confused. At least now I've made sure it is pretty clear...